src/main/java/com/spotify/confidence/SdkUtils.java

```diff
@@ -16,4 +27,105 @@ static String getSdkVersion() {
       throw new RuntimeException("Can't determine version of the SDK", e);
     }
   }
+
+  static FlagPath getPath(String str) {
+    final String regex = Pattern.quote(".");
+    final String[] parts = str.split(regex);
+
+    if (parts.length == 0) {
+      // this happens for malformed corner cases such as: str = "..."
+      log.warn("Illegal path string '{}'", str);
+      throw new GeneralError(String.format("Illegal path string '%s'", str));
+    } else if (parts.length == 1) {
+      // str doesn't contain the delimiter
+      return new FlagPath(str, List.of());
+    } else {
+      return new FlagPath(parts[0], Arrays.asList(parts).subList(1, parts.length));
+    }
+  }
+
+  static class FlagPath {
+
+    private final String flag;
+    private final List<String> path;
+
+    public FlagPath(String flag, List<String> path) {
+      this.flag = flag;
+      this.path = path;
+    }
+
+    public String getFlag() {
+      return flag;
+    }
+
+    public List<String> getPath() {
+      return path;
+    }
+  }
+
+  static Value getValueForPath(List<String> path, Value fullValue) {
```

Comment:
Can we extract this to live in the `ConfidenceFeatureProvider` or a dedicated provider utils class?